subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultFileOperations.java

```diff
@@ -73,13 +74,18 @@
     private final FileSystem fileSystem;
     private final DirectoryFileTreeFactory directoryFileTreeFactory;
 
+    @Deprecated //used by the Kotlin DSL
```

Comment:
@eskatos The Kotlin DSL will need to use the new constructor to allow this feature in Kotlin scripts.

It would be great if we could limit the API surface more, this is a pretty low-level type.